SmellDetector/SmellModel/PartialSmellDetectionReport.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Collections.Generic;
+
+namespace SmellDetector.SmellModel
+{
+    public class PartialSmellDetectionReport
+    {
+        public Dictionary<string, List<Issue>> PartialReport { get; set; }
+
+        public PartialSmellDetectionReport()
+        {
+            PartialReport = new Dictionary<string, List<Issue>>();
+        }
+
+        public void AddIssue(string id, Issue issue)
+        {
+            if (!PartialReport.ContainsKey(id)) PartialReport.Add(id, new List<Issue> { issue });
+            else PartialReport[id].Add(issue);
+        }
+
+        public void AddIssues(string id, List<Issue> issues)
+        {
+            if (!PartialReport.ContainsKey(id)) PartialReport.Add(id, issues);
```

Comment:
You have duplicate logic here. You can transform AddIssue:
`public void AddIssue(string id, Issue issue)
{
  AddIssues(id, new List<Issue> { issue });
}`

Or go the other way around and have AddIssues call AddIssue.

Also lets rename the id to be codeItemId and make it more explicit.